pkg/logs/input/file/tailer_windows.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build windows
+
+package file
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/decoder"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// setup sets up the file tailer
+func (t *Tailer) setup(offset int64, whence int) error {
+	fullpath, err := filepath.Abs(t.path)
+	if err != nil {
+		return err
+	}
+	t.fullpath = fullpath
+
+	// adds metadata to enable users to filter logs by filename
+	t.tags = t.buildTailerTags()
+
+	log.Info("Opening ", t.fullpath)
+	f, err := openFile(t.fullpath)
+	if err != nil {
+		return err
+	}
+	filePos, _ := f.Seek(offset, whence)
+	f.Close()
+
+	t.readOffset = filePos
+	t.decodedOffset = filePos
+
+	return nil
+}
+
+func (t *Tailer) readAvailable() (err error) {
+	f, err := openFile(t.fullpath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	st, err := f.Stat()
+	if err != nil {
+		log.Debugf("Error stat()ing file %v", err)
+		return err
+	}
+
+	sz := st.Size()
+	offset := t.GetReadOffset()
+	log.Debugf("Size is %d, offset is %d", sz, offset)
+	if sz == 0 {
+		log.Debug("File size now zero, resetting offset")
+		t.SetReadOffset(0)
+		t.SetDecodedOffset(0)
+	} else if sz < offset {
+		log.Debug("Offset off end of file, resetting")
+		t.SetReadOffset(0)
+		t.SetDecodedOffset(0)
+	}
+	f.Seek(t.GetReadOffset(), io.SeekStart)
```

Comment:
I think handling the file rotation in the tailer instead of in the scanner can lead to missing the last log entries of the rotated file.

Previous logic (on all platforms) was:
1. scanner detects a file rotation
2. scanner creates a new tailer on the new file, and the scanner leaves the old tailer running on the rotated file for a maximum of 60 seconds so the old tailer finishes tailing the old file, see https://github.com/DataDog/datadog-agent/blob/7.19.2/pkg/logs/input/file/scanner.go#L255-L267

This allows not missing the last log entries of the rotated log file.

Whereas new logic on Windows is:
1. scanner doesn't try to detect the log rotation, when a log rotation happens it doesn't change the tailer(s)
2. when tailer detects a file rotation, it immediately switches to the new log file, without checking first whether it was able to read everything from the rotated log file.

So I think this new logic on Windows will lead to missing the last log entries of the rotated log file.